### PR TITLE
[Mellanox] [SKU] Mellanox-SN4700-C128 SKU added

### PR DESCRIPTION
**Pull request by vivekrnv**
Signed-off-by: Vivek Reddy Karri <vkarri@nvidia.com>

#### Why I did it

Draft a new SKU for MSN-4700 Platform i.e. **Mellanox-SN4700-C128**

#### Requirements:

- **Breakout:** Port 1-32: 4x100G
- **Downlinks:** 120 (1-30)
- **Uplinks:** 8 (31-32)
- **Shared Headroom:** Enabled
- **Over Subscribe Ratio:** 1:8
- **Default Topology:** T2
- **Default Cable Length for T2:** 1500m
-  **QoS params**: The default ones defined in [qos_config.j2](https://github.com/Azure/sonic-buildimage/blob/master/files/build_templates/qos_config.j2) will be applied
-  **Small Packet Percentage:** Used 50% for traditional buffer model Note: For dynamic model, the value defined in `LOSSLESS_TRAFFIC_PATTERN|AZURE|small_packet_percentage` is used

##### Additional Details:
-  Switch Type has to be programmed as `SpineRouter` through config_db.json in `DEVICE_METADATA|localhost|type` field for the buffer values & cable lengths defined in the buffers_defaults_t2.j2 to apply on the device
- Cable Lengths Used for generating `buffer_defaults_{t0,t1,t2}.j2` values
```
 {
        "t2": {
            "up": "1500m",
            "down": "1500m",       
         },
        "t1": {
            "up": "300m",
            "down": "40m",
        },
        "t0": {
            "up": "40m",
            "down": "5m",
        }
}
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205

#### A picture of a cute animal (not mandatory but encouraged)
[syslog.gz](https://github.com/sonic-net/sonic-buildimage/files/9264243/syslog.gz)

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

**Comment by zhenggen-xu** [comment]
Can you please share the commit-id this PR was tested on?   and also the syslog when you do "config reload" the configurations with Qos/buffer enabled on this HWSKU?

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-C128/buffers_defaults_t2.j2 [hunk 1]
```diff
@@ -0,0 +1,46 @@
+
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-C128/buffers_defaults_t2.j2:1]: why do we have these extra 2 empty lines?

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-C128/buffers_defaults_t1.j2 [hunk 2]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-C128/buffers_defaults_t1.j2:1]: why do we have these extra 2 empty lines?

File: device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-C128/buffers_defaults_t0.j2 [hunk 3]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4700-r0/Mellanox-SN4700-C128/buffers_defaults_t0.j2:1]: why do we have these extra 2 empty lines?

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**